app/pages/project/workflow-selection.jsx

```diff
@@ -12,12 +12,17 @@ class WorkflowSelection extends React.Component {
     super();
     this.state = {
       error: null,
-      loadingSelectedWorkflow: true
+      loadingSelectedWorkflow: false
     };
   }
 
   componentDidMount() {
-    this.getSelectedWorkflow(this.props);
+    const { project, workflow } = this.props;
+    const linkedActiveWorkflows = project.links.active_workflows || [];
+    const workflowExistsForProject = workflow && linkedActiveWorkflows.indexOf(workflow.id) > -1;
+    if (!workflowExistsForProject) {
+      this.getSelectedWorkflow(this.props);
```

Comment:
Ok, so this only gets called if it needs to (if there isn't currently a workflow present? And this is what eliminates the second workflow request?

Reply:
Yes. Without this change, a workflow is always loaded on component mount, even if there’s already one in the Redux store.